except_star.md

```diff
@@ -234,20 +284,177 @@ except *ValueError as errors:
+#     ValueError('b')
+#   )
+```
+
+### Recursive Matching
+
+The matching of `except *` clauses against an `ExceptionGroup` is performed
+recursively. E.g.:
+
+```python
+try:
+  raise ExceptionGroup(
+    ValueError('a'),
+    TypeError('b'),
+    ExceptionGroup(
+      TypeError('c'),
+      KeyError('d')
+    )
+  )
+except *TypeError as e:
+  print(f'got some TypeErrors: {len(e)}')
+except *Exception:
+  pass
+
+# would print:
+#
+#  got some TypeErrors: 2
+```
+
+Iteration over an `ExceptionGroup` that has nested `ExceptionGroup` objects
+in it effectively flattens the entire tree. E.g.
+
+```python
+print(
+  list(
+    ExceptionGroup(
+      ValueError('a'),
+      TypeError('b'),
+      ExceptionGroup(
+        TypeError('c'),
+        KeyError('d')
+      )
+    )
+  )
+)
+
+# would output:
+#
+#   [ValueError('a'), TypeError('b'), TypeError('c'), KeyError('d')]
+```
+
+### "raise e" vs "raise"
+
+The difference between bare `raise` and a more specific `raise e` is more
+significant for exception groups than it is for regular exceptions. Consider
+the following two examples that illustrate it.
+
+Bare `raise` preserves the exception group internal structure:
+
+```python
+try:
+  raise ExceptionGroup(
+    ValueError('a'),
+    TypeError('b'),
+    ExceptionGroup(
+      TypeError('c'),
+      KeyError('d')
+    )
+  )
+except *TypeError as e:
+  raise
+
+# would crash with:
+#
+#  ExceptionGroup(
+#    ValueError('a'),
+#    TypeError('b'),
+#    ExceptionGroup(
+#      TypeError('c'),
+#      KeyError('d')
+#    )
+#  )
+```
+
+Whereas `raise e` would flatten the captured subset:
+
+```python
+try:
+  raise ExceptionGroup(
+    ValueError('a'),
+    TypeError('b'),
+    ExceptionGroup(
+      TypeError('c'),
+      KeyError('d')
+    )
+  )
+except *TypeError as e:
+  raise e
+
+# would crash with:
+#
+#  ExceptionGroup(
+#    ValueError('a'),
+#    ExceptionGroup(
+#      TypeError('b'),
+#      TypeError('c'),
+#    ),
+#    ExceptionGroup(
+#      KeyError('d')
+#    )
+#  )
 ```
 
-### Types of errors
+### "continue" and "break" in "except*"
+
+Both `continue` and `break` are disallowed in `except*` clauses, causing
+a `SyntaxError`.
+
+Due to the fact that `try..except*` block allows multiple `except*` clauses
+to run while handling one `ExceptionGroup` with multiple different exceptions
+in it, allowing one innocent `break` or `continue` in one `except*` to
+effectively silence the entire group feels very error prone.
+
+### "return" in "except*"
+
+A `return` in a regular `except` or `finally` clause means
+"suppress the exception". For example, both of the below functions would
+silence their `ZeroDivisionError`s:
+
+```python
+def foo():
+   try:
+      1 / 0
+   finally:
+      print('the sound of')
+      return
+
+def bar():
+   try:
+      1 / 0
+   except ZeroDivisionError:
+     pass
+   finally:
+      print('silence')
+      return
+
+foo()
+bar()
+
+# would print:
+#
+#   the sound of
+#   silence
+```
```

Comment:
That example is irrelevant because it's about `finally`. Please show an example (preferably a simpler one) using `return` in `except`.

Reply:
Oups, I pasted a wrong example. :( Fixing it now.